SFS_general/sphbesselh_zeros.m

```diff
@@ -53,74 +66,114 @@
 isargpositivescalar(order);
 
 
-%% ===== Configuration ===================================================
-% Method for calculating zeros
-% See https://github.com/sfstoolbox/sfs/issues/57 for a discussion
-method = 1;
+%% ===== Constants =======================================================
+TOL = 1E-15;
+MAXITER = 50;
 
 
 %% ===== Main ============================================================
-if order<86
-    if method==1
-        % Method 1 after FIXME
-        % Formula for nominator (source?)
-        B = zeros(1,order+2);
-        for n=0:order
-            B(n+1) = factorial(2*order-n)/(factorial(order-n)*factorial(n)*2^(order-n));
-        end
-        B = B(end:-1:1);
-        % Zeros
-        z = roots(B);
-        % Poles (are always zero)
-        p = zeros(order,1);
-    elseif method==2
-        % Method 2 after Pomberger 2008, p. 43
-        B = cell(order+1,1);
-        z = cell(order+1,1);
-        p = cell(order+1,1);
-        for n=0:order
-          % Recursion formula for nominator
-          B{n+1} = zeros(1,n+1);
-          for k=0:n-1
-              B{n+1}(k+1) = ((2*n-k-1)*(2*n-k)) / (2*(n-k)) * B{n}(k+1);
-          end
-          B{n+1}(n+1) = 1;
+p = zeros(order,1);  % poles are always zero
+
+if order < 2
+    z = -ones(order,1);
+    return
+end
+
+% Approximate roots of nth-order ordinary Bessel polynomial as starting points
+r = campos_zeros(order);
+
+% Zeros of nth-order ordinary Bessel polynomial are the same as for
+% exp(1/x)*K_{n+0.5}(1/x), where K_v is the modified Bessel function of second
+% kind. We, hence, define the target function and its first derivative as
+f = @(x) besselk(order+0.5, 1./x, 1);  % modified Bessel function scaled by exp
```

Comment:
I was a little bit confused by it.
Maybe it helps if we state explicitly the equation in the first comment. Something like
```Matlab
% Zeros of nth-order ordinary Bessel polynomial ... are the same as for modified
% Bessel function of second kind K_v:
% exp(1/x) * K_{n+0.5}(1/x) = ...
% Hence, we define the target function and its first derivative as
```
Where you have to fill the `...` part. Is it also second kind and then `Y`?